dvc/remote/__init__.py

```diff
@@ -21,6 +21,6 @@ def supported_url(url):
 
 def Remote(project, config):
     for r in REMOTES:
-        if r.supported(config):
+        if r.supported(config, check_dependencies=True):
             return r(project, config)
     raise DvcException('Remote \'{}\' is not supported.'.format(config))
```

Comment:
This could probably be improved by also mentioning that the optional dependency is missing.